### PR TITLE
Make test/cmd/a*.sh pass shellcheck

### PR DESCRIPTION
**Pull request by xychu**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug

/kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**:
Part of #72956
**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:
This is part of fixes under `test/cmd/*.sh` and follows the pattern in #74338
**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by xychu** [comment]
kindly ping @liggitt , PTAL when it's convenient.

**Comment by liggitt** [comment]
will defer to @BenTheElder for review

**Review by BenTheElder — APPROVED** [review]
/lgtm
needs a rebase, sorry about that :sweat_smile:

**Comment by xychu** [comment]
@BenTheElder Thanks for your review, have rebased.

**Review by BenTheElder — APPROVED** [review]
/lgtm

**Review by dixudx — APPROVED** [review]

**Comment by xychu** [comment]
Kindly ping @BenTheElder @liggitt , still need one approved label :-)

File: test/cmd/apply.sh
Comment (liggitt) [inline test/cmd/apply.sh]: `grep -q` will cause pipe errors when used this way... see https://github.com/kubernetes/kubernetes/pull/57805

**Comment by liggitt** [comment]
/hold

see comment on issues caused by this `grep -q` usage

**Comment by xychu** [comment]
@BenTheElder  @liggitt thanks for your review, all `grep -q` usage should have been fixed. PTAL.

File: test/cmd/apply.sh
Comment (liggitt) [inline test/cmd/apply.sh]: is `! grep ... <<< ...` equivalent to `! ... | grep ...`? @BenTheElder

Comment (BenTheElder) [reply test/cmd/apply.sh]: no, this is wrong. `<<<` is a here string, it will be passed to stdin. this line:
`! grep -q RollingUpdate <<< kubectl get deployments test-deployment-retainkeys -o yaml "${kube_flags[@]:?}"`

is equivilant to:

`! echo "kubectl" | grep -q RollingUpdate get deployments test-deployment-retainkeys -o yaml "${kube_flags[@]:?}"`

Comment (xychu) [reply test/cmd/apply.sh]: @liggitt @BenTheElder Updated and changed to:

`grep -q RollingUpdate <<< "$(kubectl get deployments test-deployment-retainkeys -o yaml "${kube_flags[@]:?}")"`

File: test/cmd/apply.sh
Comment (BenTheElder) [reply test/cmd/apply.sh]: dug into this a bit more, bash makes a tempfile, writes the herestring contents to it, and then execs the command with the tempfile as stdin.

`strace -e execve,open,dup2,pipe,read,write -f bash -c 'cat <<<"test"'`

**Comment by xychu** [comment]
@BenTheElder kindly ping. any further changes needed?

**Review by BenTheElder — APPROVED** [review]
/lgtm
/approve

**Comment by xychu** [comment]
@liggitt are we okay to cancel the hold :-)

**Comment by liggitt** [comment]
@BenTheElder was this reviewed really closely? Logic errors or exit code masking here can mask actual test failures and negate the coverage of this suite

**Comment by BenTheElder** [comment]
@liggit I read this over a few times and didn't spot any issues, the changes should make these variables we expect to read actually required though. It seems to be passing in CI so we should have them now and it should catch if that changes.

**Pull request merged**
